packages/form-state-manager/src/utils/manager-api.ts

```diff
@@ -18,10 +18,38 @@ const createManagerApi: CreateManagerApi = (onSubmit) => {
     unregisterField,
     getState,
     registeredFields: [],
-    fieldListeners: {}
+    fieldListeners: {},
+    active: null,
+    dirty: false,
+    dirtyFields: [],
+    dirtyFieldsSinceLastSubmit: [],
+    dirtySinceLastSubmit: false,
+    error: null,
+    hasSubmitErrors: false,
+    hasValidationErrors: false,
+    initialValues: {},
+    invalid: false,
+    modified: {},
+    modifiedSinceLastSubmit: false,
+    submitError: null,
+    submitErrors: {},
+    submitFailed: false,
+    submitSucceeded: false,
+    submitting: false,
+    touched: {},
+    valid: true,
+    validating: false,
+    visited: {}
   };
+
   function change(name: string, value?: any): any {
     state.values[name] = value;
+    state.visited[name] = true;
+    state.modified[name] = true;
+    state.modifiedSinceLastSubmit = true;
+    state.dirtySinceLastSubmit = true;
+    state.dirtyFields.push(name);
```

Comment:
This will add the same name multiple times to the array on each change. I don't think we want that. We can easily check if the field is pristine/dirty and make a set of initial changes only once. (pristine, dirty, touched, etc.)

Reply:
You are right.

But I feel that all this computation will be refactored a lot of times since now, so this just a 'placeholder'.

But it's useful to write tests, because they won't be changed.